### ISSUE TITLE
Won't reconnect if wifi is down on powerup

### ISSUE THREAD

**Issue by bneedhamia** [bneedhamia/MAX31820WeatherStation#2]
The Sketch attempts to connect to WiFi only in Setup(), and goes to an error state if it can't connect. If the WiFi hotspot is down when the station is powered up, the station won't report until the station power is manually reset.

In short, the station won't recover properly from a power outage, because when power is restored, the WiFi isn't up yet when the station powers up.

The fix: expand the state machine to include WiFi connection-related states.

**Comment by bneedhamia** [comment]
I tested the ESP8266 HTTPSRequest Sketch (https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266WiFi/examples/HTTPSRequest/HTTPSRequest.ino), which simply waits forever for the WiFi Access Point to connect. Found I can start that Sketch with the AP off, turn the AP on 10 minutes later, and the Sketch will connect to it and run.
So I want to add a state to the loop() State Machine that indicates the Sketch is waiting for the ESP8266 to connect to the Access Point. I'll also make the EEPROM timeout parameter unused, but will leave it in in case a later version needs it.
